client/dashboard/data/site-owner-transfer.ts

```diff
@@ -16,7 +16,8 @@ export async function startSiteOwnerTransfer( siteId: number, data: { new_site_o
 			calypso_origin: window.location.origin,
 		},
 		{
-			context: 'dashboard_v2',
+			// We need to update the check when modifying the base path for v2
+			context: window.location.pathname.startsWith( '/v2' ) ? 'dashboard_v2' : undefined,
```

Comment:
WDYT if we add context="dashboard_v2" from the router https://github.com/Automattic/wp-calypso/blob/75c93843603bb2b23c9b6319e6c9a63007acac38/client/dashboard/app/router.tsx#L355

And pass it to startSiteOwnerTransfer as part of data?

Reply:
Done by https://github.com/Automattic/wp-calypso/pull/104196/commits/b2cd0915ecaebd1685bb6656ec70689b51d33a9b.